index.js

```diff
@@ -55,22 +84,59 @@ async function init() {
 
 	console.log('Source:', {repo, ref, dir});
 
+	if (!navigator.onLine) {
+		updateStatus('⚠ You are offline.');
+		throw new Error('User agent is offline');
+	}
+
 	updateStatus('Retrieving directory info…');
 
+	await ensureRepoIsAccessible(repo);
+
 	const files = await listContent.viaTreesApi(`${repo}#${ref}`, decodeURIComponent(dir), localStorage.token, ref);
```

Comment:
I just noticed that this was wrong, can you fix it as well?

```suggestion
	const files = await listContent.viaTreesApi(`${repo}#${ref}`, decodeURIComponent(dir), localStorage.token);
```

There's no 4th parameter: https://github.com/fregante/list-github-dir-content/blob/v1.1.1/index.js#L35